### PR TITLE
Adds a section for user namespace mappings

### PR DESCRIPTION
**Pull request by mrunalp**
Signed-off-by: Mrunal Patel <mrunalp@gmail.com>

### CODE REVIEW

File: config-linux.md [hunk 1]
```diff
@@ -118,6 +118,21 @@ For more information, see [the man page](http://man7.org/linux/man-pages/man8/sy
 
 rlimits allow setting resource limits. The type is from the values defined in [the man page](http://man7.org/linux/man-pages/man2/setrlimit.2.html). The kernel enforces the soft limit for a resource while the hard limit acts as a ceiling for that value that could be set by an unprivileged process.
 
+## Linux user namespace mappings
+
+```
+    "userMappings": [
```
Comment (LK4D4) [inline config-linux.md]: btw I wonder why we don't use capital-first camelcase.

**Comment by LK4D4** [comment]
How about groupMappings?

File: config-linux.md [hunk 2]
```diff
@@ -118,6 +118,21 @@ For more information, see [the man page](http://man7.org/linux/man-pages/man8/sy
...
+        {
+            "from"  : 1000,
+            "to"    : 0,
+            "count" : 10
```
Comment (LK4D4) [inline config-linux.md]: Not sure about `count`, it called
```
The length of the range of user IDs
```
in man page.

File: config-linux.md (hunk 1 above)
Comment (mrunalp) [inline config-linux.md]: Just matching what we have in spec_linux.go. We could even go with user_mappings.

Comment (crosbymichael) [inline config-linux.md]: in proper json you don't capitalize the first letter

File: config-linux.md (hunk 2 above)
Comment (mrunalp) [inline config-linux.md]: That's how we have it today in the spec. We could either go with how it is in the kernel or what we have in golang.

**Comment by mrunalp** [comment]
Added gidMappings and changed from userMappings to uidMappings.

**Comment by LK4D4** [comment]
LGTM

**Comment by cbinion46511** [comment]
???

**Comment by crosbymichael** [comment]
@mrunalp do you also want to change the types now that they are in the repo?

**Comment by mrunalp** [comment]
@crosbymichael Yep, will update this in a few.

**Comment by mrunalp** [comment]
@crosbymichael Actually spec_linux.go matches what we have here unless I am misreading it..

File: config-linux.md [hunk 3]
```diff
@@ -118,6 +118,27 @@ For more information, see [the man page](http://man7.org/linux/man-pages/man8/sy
...
+    "gidMappings": [
+        {
+            "from"  : 1000,
+            "to"    : 0,
+            "count" : 10
+        }
+    ]
+```
+
+uid/gid mappings describe the user namespace mappings from the host to the container. *from* is the starting uid/gid on the host to be mapped to *to* which is the starting uid/gid in the container and *count* refers to the number of ids to be mapped. The linux kernel has a limit of 5 such mappings that could be specified.
```
Comment (jonboulle) [inline config-linux.md]: The Linux kernel has a limit of 5 such mappings that can be specified.

Comment (mrunalp) [inline config-linux.md]: Updated.

Comment (jonboulle) [inline config-linux.md]: nit, you missed linux -> Linux

Comment (mrunalp) [inline config-linux.md]: done.

**Comment by wking** [comment]
I'm proposing changes to the field names in #63 and
opencontainers/runc#98.  If those branches look promising, we'll want
to reroll this branch to match the new names.

**Comment by crosbymichael** [comment]
LGTM

**Pull request merged**